x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/index.test.tsx

```diff
@@ -94,8 +117,303 @@ describe('SystemPrompt', () => {
+        </TestProviders>
+      );
+      userEvent.click(screen.getByTestId('edit'));
+      userEvent.click(screen.getByTestId(TEST_IDS.ADD_SYSTEM_PROMPT));
+
+      expect(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.ID)).toBeVisible();
+
+      userEvent.type(
+        within(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_SELECTOR)).getByTestId('comboBoxInput'),
+        `${customPromptName}[Enter]`
+      );
+
+      userEvent.type(
+        screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.PROMPT_TEXT),
+        customPromptText
+      );
+
+      userEvent.click(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.SAVE));
+
+      await waitFor(() => {
+        expect(mockUseAssistantContext.setAllSystemPrompts).toHaveBeenCalledTimes(1);
+        expect(mockUseAssistantContext.setAllSystemPrompts).toHaveBeenNthCalledWith(1, [
+          mockSystemPrompt,
+          {
+            id: customPromptName,
+            content: customPromptText,
+            name: customPromptName,
+            promptType: 'system',
+          },
+        ]);
+        expect(screen.queryByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.ID)).toBeFalsy();
+      });
+    });
+
+    it('should save new prompt as a default prompt', async () => {
+      const customPromptName = 'custom prompt';
+      const customPromptText = 'custom prompt text';
+      render(
+        <TestProviders>
+          <SystemPrompt conversation={BASE_CONVERSATION} />
+        </TestProviders>
+      );
+      userEvent.click(screen.getByTestId('edit'));
+      userEvent.click(screen.getByTestId(TEST_IDS.ADD_SYSTEM_PROMPT));
+
+      expect(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.ID)).toBeVisible();
+
+      userEvent.type(
+        within(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_SELECTOR)).getByTestId('comboBoxInput'),
+        `${customPromptName}[Enter]`
+      );
+
+      userEvent.type(
+        screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.PROMPT_TEXT),
+        customPromptText
+      );
+
+      userEvent.click(
+        screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.TOGGLE_ALL_DEFAULT_CONVERSATIONS)
+      );
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.TOGGLE_ALL_DEFAULT_CONVERSATIONS)
+        ).toBeChecked();
+      });
+
+      userEvent.click(screen.getByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.SAVE));
+
+      await waitFor(() => {
+        expect(mockUseAssistantContext.setAllSystemPrompts).toHaveBeenCalledTimes(1);
+        expect(mockUseAssistantContext.setAllSystemPrompts).toHaveBeenNthCalledWith(1, [
+          {
+            ...mockSystemPrompt,
+            isNewConversationDefault: false,
+          },
+          {
+            id: customPromptName,
+            content: customPromptText,
+            name: customPromptName,
+            promptType: 'system',
+            isNewConversationDefault: true,
+          },
+        ]);
+        expect(screen.queryByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.ID)).toBeFalsy();
```

Comment:
Consider asserting `.not.toBeInTheDocument()` as an alternative to `.toBeFalsy()`, i.e.

```ts
expect(screen.queryByTestId(TEST_IDS.SYSTEM_PROMPT_MODAL.ID)).not.toBeInTheDocument();
```

if the former feels like the truthier intent of the assertion in this, and similar tests.

Reply:
Yes.. I intended to change it but forgot about it. Thanks for catching it.